app/controllers/ConfirmationController.scala

```diff
@@ -40,10 +42,10 @@ class ConfirmationController @Inject()(
 
   def onPageLoad(ern: String, draftId: String): Action[AnyContent] =
     authorisedDataRequest(ern, draftId) { implicit request =>
-      request.session.get(SUBMISSION_RECEIPT_REFERENCE) match {
-        case Some(reference) =>
-          Ok(view(reference))
-        case None =>
+      (request.session.get(SUBMISSION_RECEIPT_REFERENCE), request.userAnswers.get(DeclarationPage)) match { //TODO Update Submission Reference
+        case (Some(submissionReference), Some(submissionTimestamp)) =>
+          Ok(view(submissionReference, submissionTimestamp.toLocalDate))
+        case req =>
           logger.warn("[onPageLoad] Could not retrieve submission receipt reference from Users session")
           BadRequest(errorHandler.badRequestTemplate)
```

Comment:
Might as well redirect to Journey Recovery, there's nothing the user can do at this point to fix their error here.